Extensions/Source/ShortNumberFormatter.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Copyright © 2019 Shakuro. All rights reserved.
+//
+
+import Foundation
+
+public final class ShortNumberFormatter {
+
+    private enum Constant {
+        static let suffixes: [String] = [ "", "k", "M", "G", "T", "P", "E" ]
+        static let step: Double = 1000
+        static let digitsInStep: Int = 3
+    }
+
+    let numberFormatter: NumberFormatter
+
+    ///
+    /** - Parameter numberFormatter: NumberFormatter instance or nil,
+     in case of nil default number formatter (maximumFractionDigits = 2, minimumFractionDigits = 0, roundingMode = .down ) will be created atumatically
+     */
+    public init(numberFormatter: NumberFormatter? = nil) {
+        if let actualFormatter = numberFormatter {
+            self.numberFormatter = actualFormatter
+        } else {
+            let formatter: NumberFormatter = NumberFormatter()
+            formatter.locale = Locale.current
+            formatter.maximumFractionDigits = 2
+            formatter.minimumFractionDigits = 0
+            formatter.roundingMode = .down
+            self.numberFormatter = formatter
+        }
+    }
+
+    /// Transforms double value to string with SI decimal prefix, 1000 -> 1k
+    ///
+    /// - Parameter doubleValue: The value to transform
+    /// - Returns: A formatted string
+    public func string(for doubleValue: Double) -> String {
+        let value: Double = abs(doubleValue)
+        let suffixes = Constant.suffixes
+        let suffixIndex: Int
+        let shortValue: Double
+        if value < Constant.step {
+            suffixIndex = 0
+            shortValue = value
+        } else {
+            let digitCount: Double = log10(value)
+            if digitCount.isFinite {
+                let maxIndex = suffixes.count - 1
+                let maxDigitCount = Double(Constant.digitsInStep * maxIndex)
+                suffixIndex = digitCount < maxDigitCount ? Int(digitCount / Double(Constant.digitsInStep)) : maxIndex
+                shortValue = value/pow(Constant.step, Double(suffixIndex))
```

Comment:
would it better to use `doubleValue` (to avoid later `let resultValue = doubleValue < 0 ? -shortValue : shortValue`)?

Reply:
In any case abs value is required for comparison with Constant.step and to compute logarithm. And sign is not  involved in calculations, it is constant. so i think current solution is better.